### PR TITLE
Assembly metadata

### PR DESCRIPTION
**Pull request by joverlee521**
Create an "assembly-exporter" role and a new shipping view `shipping.genome_submission_metadata_v1`. 

For context, this is how I intend to use the view for submissions:
```sql
    create temp table batch(nwgc_id text, sfs_sample_barcode text);
    \copy batch ("$columns") from '"$1"' csv header

    create temp view batch_export as (
        select
            nwgc_id,
            sfs_sample_identifier,
            sfs_sample_barcode,
            collection_date,
            swab_type,
            source,
            puma,
            state
        from batch
        left join shipping.genome_submission_metadata_v1 using (sfs_sample_barcode)
        order by nwgc_id
    );

    \copy (select * from batch_export) to pstdout with csv header
```

### CODE REVIEW

**Comment by joverlee521** [comment]
[New shipping view for exporting metadata for consensus genome submissions](https://trello.com/c/UxxrNeUF/1037-new-shipping-view-for-exporting-metadata-for-consensus-genome-submissions)

File: schema/deploy/shipping/views.sql [hunk 1]
```diff
@@ -2083,6 +2084,46 @@ comment on view shipping.flu_assembly_jobs_v1 is
...
+        sample.collected::date as collection_date,
+        case sample.details ->> 'swab_type'
+          when 'ans' then 'Anterior nasal swab'
+          when 'mtb' then 'Mid-turbinate nasal swab'
+          when 'np' then 'Nasopharyngeal swab'
+          else null
+        end as swab_type,
+        case
+          when identifier_set.name in ('collections-scan', 'collections-scan-kiosks') then 'SCAN'
+          else 'SFS'
```
Comment (jccraft-uw) [inline schema/deploy/shipping/views.sql:2101]: If not SCAN then SFS ... this won't hold forever. Apple, for example. 
There's also discussion about us not including Apple data with other data. That's still being discussed, so we don't have to filter anything just yet. I can imagine having to exclude the two Apple collection sets from this view.

**Review by jccraft-uw — APPROVED** [review]

File: schema/deploy/shipping/views.sql (hunk 1 above)
Comment (tsibley) [reply schema/deploy/shipping/views.sql:2101]: SFS/SCAN is the org, not the study. I think all current and future studies are happening under the aegis of SFS.

File: schema/deploy/shipping/views.sql [hunk 2]
```diff
@@ -2083,6 +2084,46 @@ comment on view shipping.flu_assembly_jobs_v1 is
 alter view shipping.flu_assembly_jobs_v1 owner to "view-owner";
 
 
+create or replace view shipping.genome_submission_metadata_v1 as
+
+    select
+        sample.identifier as sfs_sample_identifier,
+        sampleid.barcode as sfs_sample_barcode,
```
Comment (tsibley) [inline schema/deploy/shipping/views.sql:2091]: Little weird to me to have these called `sfs_…` when they're not all SFS.

**Review by tsibley — APPROVED** [review]
> For context, this is how I intend to use the view for submissions:
>
>     create temp table batch(nwgc_id text, sfs_sample_barcode text);
>     \copy batch ("$columns") from '"$1"' csv header
>     …
>         left join shipping.genome_submission_metadata_v1 using (sfs_sample_barcode)


I'd suggest using `citext` for `sfs_sample_barcode` in the temp view so that the join on it with the same column from `genome_submission_metadata_v1` is robust to case.

File: schema/deploy/shipping/views.sql (hunk 1 above)
Comment (joverlee521) [reply schema/deploy/shipping/views.sql:2101]: Everything is still under the SFS org, the separation of SCAN is to add extra authors from PHSKC. 
I've asked Trevor on [Slack](https://seattle-flu-study.slack.com/archives/C01MQTSJH44/p1614717832026400) to verify that all other SFS studies can use the general SFS authors list.

**Comment by joverlee521** [comment]
> I'd suggest using citext for sfs_sample_barcode in the temp view so that the join on it with the same column from genome_submission_metadata_v1 is robust to case.

Ah, thank you for the suggestion! I'll change the type to citext in the assembly script.

**Review by tsibley — APPROVED** [review]

**Comment by joverlee521** [comment]
Specifically named the schools collections to "Snohomish Schools" since the new Yakima schools study is going to be a completely different/separate study. 

Also added `coalesce` for the collection date to fall back on the encountered date if the collected date is null for some reason. I realized this is needed for retro samples that do not have a date on the sample tubes, but have an encountered date from the clinical metadata.

**Review by kschwabefry — APPROVED** [review]

**Pull request merged**